core/src/main/java/pl/agh/edu/screen/main/MainScreen.java

```diff
@@ -1,55 +1,54 @@
-package pl.agh.edu.screen;
+package pl.agh.edu.screen.main;
 
 import com.badlogic.gdx.Gdx;
-import com.badlogic.gdx.InputMultiplexer;
 import com.badlogic.gdx.Screen;
 import com.badlogic.gdx.scenes.scene2d.Stage;
-import com.badlogic.gdx.scenes.scene2d.ui.Container;
-import com.badlogic.gdx.scenes.scene2d.ui.Skin;
 import com.badlogic.gdx.scenes.scene2d.ui.Stack;
 import com.badlogic.gdx.scenes.scene2d.ui.Table;
 
 import pl.agh.edu.GdxGame;
-import pl.agh.edu.actor.GameSkin;
 import pl.agh.edu.actor.component.background.InfinityBackground;
 import pl.agh.edu.actor.component.button.OptionButton;
+import pl.agh.edu.actor.component.modal.event.EventWrapper;
+import pl.agh.edu.actor.component.modal.options.OptionsWrapper;
 import pl.agh.edu.actor.component.navbar.NavbarBottom;
 import pl.agh.edu.actor.component.navbar.NavbarTop;
 import pl.agh.edu.actor.frame.FrameStack;
-import pl.agh.edu.actor.frame.OptionFrame;
 import pl.agh.edu.actor.shader.BlurShader;
 import pl.agh.edu.actor.utils.resolution.ResolutionChangeListener;
 import pl.agh.edu.actor.utils.resolution.ResolutionManager;
 import pl.agh.edu.config.GraphicConfig;
-import pl.agh.edu.model.time.Time;
 
 public class MainScreen implements Screen, ResolutionChangeListener {
-	public final FrameStack frameStack = new FrameStack();
-	private final Skin skin = GameSkin.getInstance();
-	private final Stack stack = new Stack();
-	private final Container<OptionFrame> optionFrameContainer = new Container<>();
-	private final Table table = new Table();
-	private boolean isOptionsOpen = false;
-	private final OptionFrame optionFrame = new OptionFrame(this::closeOptions);
+	private final GdxGame game;
 	private final Stage mainStage = new Stage(GraphicConfig.getViewport());
 	private final Stage middleStage = new Stage(GraphicConfig.getViewport());
 	private final Stage topStage = new Stage(GraphicConfig.getViewport());
-	private final BlurShader blurShader = new BlurShader(mainStage);
-	private final InputMultiplexer inputMultiplexer = new InputMultiplexer(mainStage);
+	public final FrameStack frameStack = new FrameStack();
+	private final BlurShader blurShader = new BlurShader(mainStage, middleStage);
+	private final MainScreenInputAdapter inputMultiplexer = new MainScreenInputAdapter(mainStage);
+	public final OptionsWrapper optionsWrapper = new OptionsWrapper(inputMultiplexer, blurShader, mainStage, topStage);
 	private final InfinityBackground infinityBackground = new InfinityBackground("night-city");
+	private final EventWrapper eventWrapper = new EventWrapper(inputMultiplexer, blurShader, mainStage, topStage);
 
 	public MainScreen(GdxGame game) {
+		this.game = game;
+		game.engine.eventHandler.setEventHandlerFunction(eventWrapper::showEvent);
```

Comment:
nie zbyt to ładne, ale lepszego rozwiązania nie widzę

Reply:
same